### PR TITLE
Stopped using lodash for throttle/debounce

### PR DESCRIPTION
**Pull request by wheresrhys**
@AlbertoElias @onishiweb 

Installing lodash, with its weird version numbers, occasionally causes builds to break, and it's quite easy to get rid of

### CODE REVIEW

**Comment by onishiweb** [comment]
LGTM, will let @AlbertoElias check it over before merging

**Comment by wheresrhys** [comment]
Note, I haven't actually tested this - just a quick proof of concept so far

**Comment by onishiweb** [comment]
Ok, have added do not merge to the title so we have time to test it

**Comment by onishiweb** [comment]
Updated this PR with tests now and all seems to work well. @AlbertoElias would you mind taking a look before it's merged.

**Comment by wheresrhys** [comment]
you're making me look lazy :grin: thanks

File: karma.conf.js [hunk 1]
```diff
@@ -74,6 +74,9 @@ module.exports = function(config) {
 							'imports?define=>false'
 						]
 					}
+				],
+				noParse: [
+					/\/sinon\.js/,
```
Comment (AlbertoElias) [inline karma.conf.js]: doesn't the node_modules exclude do this?

File: test/viewport.test.js [hunk 2]
```diff
@@ -11,7 +10,55 @@ function isPhantom() {
...
+	before(function() {
+		clock = sinon.useFakeTimers();
+	});
+
+	after(function() {
+		clock.restore();
+	});
+
+	it('should throttle callback to every 100ms', function(done) {
+		const callback = sinon.spy();
```
Comment (AlbertoElias) [inline test/viewport.test.js]: shouldn´t this be a stub instead of a spy?

File: test/viewport.test.js [hunk 3]
```diff
@@ -11,7 +10,55 @@ function isPhantom() {
...
+		const throttled = utils.throttle(callback, 100);
+
+		throttled();
```
Comment (AlbertoElias) [inline test/viewport.test.js]: To properly test it, you would need to call throttled more than once

**Comment by AlbertoElias** [comment]
@onishiweb left a couple comments, looking good 👍

File: karma.conf.js (hunk 1 above)
Comment (onishiweb) [inline karma.conf.js]: I'll try removing it but I copied this from the `o-header` Karma config... It was also in the issue on Webpack that I found the solution to.

File: test/viewport.test.js (hunk 3 above)
Comment (onishiweb) [inline test/viewport.test.js]: Updated now, for some reason the first time I tried that it didn't work as expected. Now it does.

**Comment by onishiweb** [comment]
Updated the tests with those two comments

File: karma.conf.js (hunk 1 above)
Comment (AlbertoElias) [inline karma.conf.js]: So does it work with it being removed? Yeah, sinon was a pain, forgot about it

File: test/viewport.test.js [hunk 4]
```diff
@@ -33,6 +33,14 @@ describe('o-viewport utils', function () {
 		clock.tick(1);
 		expect(callback.callCount).to.be(1);
 
+		throttled();
```
Comment (AlbertoElias) [inline test/viewport.test.js]: Oh, what I meant is that the throttled calls should go one after the other, and then only be executed after those initial 100ms

Comment (onishiweb) [inline test/viewport.test.js]: oh ok, cool. Will update and test that now

File: karma.conf.js (hunk 1 above)
Comment (onishiweb) [inline karma.conf.js]: Yeh it works with it being removed to. 

Had to upgrade Sinon to use the pre-release version so the faketimers would be included with the webpack fix. Fingers crossed it gets properly released soon.

**Comment by AlbertoElias** [comment]
lgtm!

**Pull request merged**